Updates to business acumen competency and adding explicit note for ML engineer section (#19)
GoProperly/organization: competencies/roles/DataScience.md

            <strong>Business Alignment - "Good Questions"</strong>
        </td>
        <td><ul>
            <li>Able to assess business questions and determine whether they are precise enough to develop an analysis or model.</li>
            <li>Able to participate in team discussions regarding analysis and capture key requirements. Capable of translating business questions to concise technical asks.</li>
            <li>Effective at soliciting feedback on the effectiveness of analyses.</li>
        </ul></td>
        <td><ul>
            <!--- Proficient  -->

>             <li>Active participation and outreach in explorating an ambiguous problem space, mapping to technical space, and  effectively communicating implications of analyses or technical methods.</li>

Reviewer: Can you fix the typo here too please? `explorating`

Author: Explorating is an actual word, but can change it to exploring if it's more clear

            <li>Able to participate in business level objectives discussions and develop testable data hypotheses.</li>

>             <li>Systematically approaches analyses with questions of effectiveness and identify adjacent opportunities or questions;

Reviewer: ```suggestion
            <li>Systematically approaches analyses with questions of effectiveness and identifies adjacent opportunities or questions;
```

                systematically checks understanding of problem and relevance of solutions.</li>
        </ul></td>
        <td><ul>
            <!--- Authority -->

>             <li>Effective at understanding when a problem space has been sufficiently explorated and communicating the limitations of continuing to explorate</li>

Reviewer: This competency grid is starting to make me question whether `explorate` is a real word or not 😆

>             <li>Effective at understanding when a problem space has been sufficiently explorated and communicating the limitations of continuing to explorate</li>

Reviewer: ```suggestion
            <li>Effective at understanding when a problem space has been sufficiently explored and communicating the limitations of continuing to explore</li>
```

>             <li>Effective thought partner for decisionmakers, improving their knowledge and critical thinking</li>

Reviewer: decision makers

>             <li>Effective thought partner for decisionmakers, improving their knowledge and critical thinking</li>

Reviewer: ```suggestion
            <li>Effective thought partner for decision makers, improving their knowledge and critical thinking</li>
```

>             <li>Explorating an ambiguous problem space, mapping to technical space, and communicating implications effectively and impactfully. Aligned to business strategy and value add</li>

Reviewer: ```suggestion
            <li>Exploring an ambiguous problem space, mapping to technical space, and communicating implications effectively and impactfully. Aligned to business strategy and value add</li>
```

            <li>Helps to organize the business-and-data alignment processes and uncover areas for improvement</li>

>             <li>Systematically approaches questions of effectiveness and identify adjacent opportunities or questions; develops ideas for improving overall data program in understanding of problem and relevance of solutions</li>

Reviewer: ```suggestion
            <li>Systematically approaches questions of effectiveness and identifies adjacent opportunities or questions; develops ideas for improving overall data program in understanding of problem and relevance of solutions</li>
```
